### ISSUE TITLE
Fix functions not being able to use variables defined in containing scopes

### ISSUE THREAD

**Issue by mikhail729** [Ralakus/graviton#10]

**Comment by AlexCouch** [comment]
Could we get a little bit more information? Do you mean in outer local scopes? Like if I create a local variable and try to reference it from inside a conditional branch, it can't be used?

**Comment by mikhail729** [comment]
@AlexCouch Yeah, that's the issue. Lets say you define a function and use a variable outside of it
```rust
let x = 5;

let add = (y: I32) {
    x + y
};

print(add(5));
```
Then `add` won't get the value of x and it'd just be zero no matter what it is, same would apply if it were like this
```rust
let main = () -> Nil {
    let x = 5;

    let add = (y: I32) {
        x + y
    };

    print(add(5));
};

main();```

In both cases they'd just print 5 even though it should print 10 in both cases

**Comment by noxabellus** [comment]
Those two examples seem to be doing different things, to me. The first one is accessing a global, and the second one is accessing a local variable of `main`. That makes the second example a closure, while the first one is still just a regular function. Implementing closures like the second example is going to be much more work and fundamentally different than just allowing functions to access globals like the first.
